public-api/src/main/java/org/pmiops/workbench/publicapi/DataBrowserController.java

```diff
@@ -243,10 +252,21 @@ public ConceptAnalysis apply(ConceptAnalysis ca) {
     @Override
     public ResponseEntity<ConceptListResponse> getAdvancedConceptSearch(SearchConceptsRequest searchConceptsRequest){
 
-        Integer maxResults=searchConceptsRequest.getMaxResults();
-        if(maxResults == null){
-            maxResults = 25;
+        Integer maxResults = searchConceptsRequest.getMaxResults();
+        if(maxResults == null || maxResults == 0){
+            maxResults = Integer.MAX_VALUE;
         }
+
+        if(searchConceptsRequest.getQuery() == null || searchConceptsRequest.getQuery().isEmpty()){
+            List<Concept> concepts = conceptDao.findConceptsOrderedByCount();
+            ConceptListResponse response = new ConceptListResponse();
+            if(maxResults < concepts.size()){
```

Comment:
I'm no java expert -- but this code looks to me like the limit should be put on the mysql query in findAllConceptsOrderedByCount.. See above comment on the conceptDao. Not sure if Java would return all 4million concepts and then filter that list.  Probably not.

Reply:
Yeah, this should be a parameter to findConceptsOrderedByCount.